packages/fast-components/src/text-field/text-field.ts

```diff
@@ -22,61 +23,127 @@ export class TextField extends FastElement {
             : this.classList.remove("filled");
     }
 
-    @attr({ attribute: "required", mode: "boolean" })
-    public required: boolean;
-    private requiredChanged(): void {
-        this.required
-            ? this.classList.add("required")
-            : this.classList.remove("required");
-    }
-
     @attr({ mode: "boolean" })
-    public readonly: boolean;
-    private readonlyChanged(): void {
-        this.readonly
+    @attr({ attribute: "readonly", mode: "boolean" })
```

Comment:
If you don't provide an attribute name, it will be derived from lowercasing the property name. So, technically you don't need to specify this one.